src/main/java/no/ntnu/mycbr/rest/AmalgamationFunctions.java

```diff
@@ -13,19 +12,22 @@
 
 public class AmalgamationFunctions {
 
-    private final List<String> amalgamationFunctions = new LinkedList<>();
-
-
+    private Concept concept = null;
+    private final List<String> amalgamationFunctionIDs = new LinkedList<>();
+    
     public AmalgamationFunctions(String conceptID) {
-
-        Project project = App.getProject();
-        Concept concept = project.getConceptByID(conceptID);
-        List<AmalgamationFct> amalgamationFunctions = concept.getAvailableAmalgamFcts();
+        concept = App.getProject().getConceptByID(conceptID);
+        setAmalgamationFunctionIDs();
```

Comment:
It is a good practice to avoid calling class methods in constructors unless necessary.